plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAttributeModifiers.java

```diff
@@ -81,27 +84,64 @@ public static MapTag mapify(AttributeModifier modifier) {
         result.putObject("name", new ElementTag(modifier.getName()));
         result.putObject("amount", new ElementTag(modifier.getAmount()));
         result.putObject("operation", new ElementTag(modifier.getOperation()));
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+            result.putObject("slot_group", new ElementTag(modifier.getSlotGroup().toString(), true));
+        }
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
+            result.putObject("key", new ElementTag(Utilities.namespacedKeyToString(modifier.getKey()), true));
+        }
+        // TODO: remove/deprecate these two
         result.putObject("slot", new ElementTag(modifier.getSlot() == null ? "any" : modifier.getSlot().name()));
-        if (NMSHandler.getVersion().isAtMost(NMSVersion.v1_20)) {
+        try {
             result.putObject("id", new ElementTag(modifier.getUniqueId().toString()));
         }
+        catch (Exception ignored) {}
         return result;
     }
 
-    public static AttributeModifier modiferForMap(Attribute attr, MapTag map) {
-        ElementTag name = map.getElement("name");
+    public static AttributeModifier modiferForMap(Attribute attr, MapTag map, TagContext context) {
         ElementTag amount = map.getElement("amount");
         ElementTag operation = map.getElement("operation");
-        ElementTag slot = map.getElement("slot", "any");
-        // TODO: MC 1.21: ID and name are now the same internal value.
-        ElementTag id = map.getElement("id");
-        UUID idValue;
         double amountValue;
         AttributeModifier.Operation operationValue = operation.asEnum(AttributeModifier.Operation.class);
         if (operationValue == null) {
             Debug.echoError("Attribute modifier operation '" + operation + "' does not exist.");
             return null;
         }
+        try {
+            amountValue = Double.parseDouble(amount.toString());
+        }
+        catch (NumberFormatException ex) {
+            Debug.echoError("Attribute modifier amount '" + amount + "' is not a valid decimal number.");
+            return null;
+        }
+        ElementTag key = map.getElement("key");
+        boolean is1_21 = NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21);

```

Comment:
what

Reply:
Oops, leftover from when that was checked a bunch - removed